### PR TITLE
Cleanup issues identified by SonarQube

### PR DESCRIPTION
**Pull request by smithkm**
Switch to using try-with-resources where opportunities have been identified by running static analysis with SonarQube.

This is all refactoring so there are no unit tests.

### CODE REVIEW

**Comment by aaime** [comment]
Oooh, sonarqube... is it something available also for geotools and geoserver?

**Comment by smithkm** [comment]
Boundless is using it for Boundless Server and its dependencies so yes it is being run on the other projects.  This was just the result of exploring how it works on my part.  I raised the idea of opening something to the community but the way we're currently using it that doesn't seem viable since it has a bunch of Boundless internal projects.

SonarQube rates the technical debts of the three projects at 1.5 kdays for GeoTools, 606 days for GeoServer, and 68 days for GWC.

File: geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/UsageStatsMonitor.java [hunk 1]
```diff
@@ -82,24 +82,32 @@ public void shutDown() {
...
+                                    + (attempts * seconds)
+                                    + " seconds. Attempt "
+                                    + attempts
+                                    + " of "
+                                    + maxAttempts
+                                    + "...";
+                    log.warn(message);
+                    if (attempts == maxAttempts) {
+                        throw new RuntimeException(message, e);
+                    }
```
Comment (tbarsballe) [inline geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/UsageStatsMonitor.java]: This whole block seems entirely identical to the one in QuotaUpdatesMonitor, above. Can it be moved to a shared superclass?
If not then perhaps update the error message to differentiate the two?

File: geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3Ops.java [hunk 2]
```diff
@@ -414,7 +414,7 @@ public Long call() throws Exception {
                         String.format(
                                 "S3 bulk delete aborted for '%s/%s'. Will resume on next startup.",
                                 bucketName, prefix));
-                return null;
+                throw e;
```
Comment (tbarsballe) [inline geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3Ops.java]: Perhaps this is whats causing the AppVeyor test failure? Did you check how null results were handled from where this gets called?

File: geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/UsageStatsMonitor.java (hunk 1 above)
Comment (smithkm) [reply geowebcache/diskquota/core/src/main/java/org/geowebcache/diskquota/UsageStatsMonitor.java]: Refactored to put most of the duplicate code in AbstractMonitor

File: geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3Ops.java (hunk 2 above)
Comment (smithkm) [reply geowebcache/s3storage/src/main/java/org/geowebcache/s3/S3Ops.java]: It seems to be one of the same errors occurring on other PRs

**Pull request merged**